packages/calypso-e2e/src/lib/utils/get-test-account-by-group.ts

```diff
@@ -0,0 +1,14 @@
+/**-
+ *
+ */
+export function getTestAccountByJestGroup() {
+	const env = process.env;
+
+	if ( env.JEST_GROUP_JETPACK ) {
+		return 'jetpackUser';
+	}
+	if ( env.JEST_GROUP_CALYPSO_PR ) {
+		return 'simpleSitePersonalPlanUser';
+	}
+	return 'defaultUser';
```

Comment:
This approach is very simple because as noted in the PR description we currently use only one group per run in TeamCity. While it is possible that an engineer would want to run multiple jest group locally, resource constraints mean this is rather unlikely.

In other words, if we encounter a scenario in the future where a TeamCity build configuration may specify two Jest groups this function will only ever return one test account key.

Another issue with this implementation is that it adds yet another location where both the test account key and the Jest group name has to be updated should something change.

Reply:
I'm not 100% sold on this implementation myself and would like a second opinion on this. Originally, I had a more condensed `if/else` logic in the test file itself - similar to how we had the ternary for the `gutenbergEdgeUser` - and perhaps going back to that would be a better move?

Reply:
Pinging @fullofcaffeine so he can see and weigh in!

Reply:
Hmm, this is a tricky one!

While this is slick, I think my current inclination is that we should not couple the jest-runner-group and the user. While they can be related, I think they each serve a distinct purpose that might serve us well to keep separate. The way I see it...
- `jest-runner-groups` only answers the question of "which test files should be run together".
- We answer the question of "which user should we use" based on ENV variables and script specific variables.

Practically speaking, this means that we expose an ENV variable called `TARGET_JETPACK_SITES` and configure the jetpack TC builds to not only target the `jetpack` jest-runner-group but also to set `TARGET_JETPACK_SITES=1`.

In the future, if we ever had to step away from `jest-runner-groups`, I think the transition would be a lot more seamless if we didn't have to pull logic related to that from our scripts and utilities. 

Thoughts?

Reply:
The point you raise about not coupling to a `jest-runner-groups` is super legit. As much as I don't really like it, setting our own environment variable in TeamCity could be the simplest and most maintainable solution here.

Reply:
Made the changes in b7129a5603c62da5534497240679aaec332646a4. What do you think?

Reply:
>Made the changes in https://github.com/Automattic/wp-calypso/commit/b7129a5603c62da5534497240679aaec332646a4. What do you think?

Yeah, I definitely prefer this solution personally, I think that looks really nice! Thanks for doing that! 😄 

If we're going to commit fully to that route, should we...
1. Add it to the `envVariables` (and all their fancy typing and default handling)
2. Make a `jetpack` [variant](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/utils/get-test-account-by-feature.ts#L13) and update the feature tables?

Reply:
> jest-runner-groups only answers the question of "which test files should be run together".
> We answer the question of "which user should we use" based on ENV variables and script specific variables.

That's how I understand it, too. I mean, at first sight, they could look like a feasible way to select accounts for tests, but they lack the flexibility of TC env params:
1. They're hardcoded in the shell script that's part of the TC build. You can't change them per build run like you can with `envs`
2. AFAIK, they're not ideal If we want to support more complex account combination criteria (i.e jetpack on Gutenberg edge) as we'd need to pass `@gutenberg-edge @jetpack` or `@gutenberg @jetpack @edge`. This would require us to make the actual build in TC more specific -- and we'd need to pass this to `jets` when calling it from the build shell script.

I think they more naturally map to the `variant` attribute of the `FeatureKey` (but I personally prefer we stick to env variables like you did [here](https://github.com/Automattic/wp-calypso/commit/b7129a5603c62da5534497240679aaec332646a4).
). I see the `variant` key as the "main flavor" of the test, and the other attributes as toppings. So for example, in the case of JetPack, `variant` would be `jetpack`. We'd then have a criterion like this:

Looking at the way it's implemented now, It looks like the best way to represent `jetpack` is not by variant, but by a flag with the same name so that we're able to declare the following criteria:

```
{jetpack: true, accountName: `jetpackUser` }
{jetpack: false, accountName: `defaultUser` }
```
In fact, we can't use `variant` as it's not mutually exclusive to Jetpack as it currently is used to define a `siteEditor`-specific test, and we could in theory have a jetpack site-editor test :), i.e:

```
{jetpack: false, accountName: `defaultUser`, variant: `siteEditor` }
```

Assuming the Jetpack build only needs to work with this specific account/site and doesn't need to support any other combinations for now, we could call it `getTestAccountByFeature` like this:

```js
const accountName = getTestAccountByFeatureKey( { jetpack: !! envVariables.TARGET_JETPACK } )
```

Reply:
Actually, I think the `variant: jetpack` would still be needed in order to "group" these criteria, otherwise, semantically, `{jetpack: false, accountName: 'defaultUser' }` would apply to anything else that's not jetpack, which is confusing and wrong. So, I guess it'd be best to have an additional optional `group` attribute and then define the rules as so:

```js
{jetpack: true, group: 'jetpack', accountName: 'jetpackUser' }
{jetpack: false, group: 'jetpack',  accountName: 'defaultUser' }
```

Then we'd call it as:

```js
const accountName = getTestAccountByFeatureKey( { jetpack: !! envVariables.TARGET_JETPACK, group: 'jetpack' } );
```

Reply:
By the way, I'm not suggesting you do this as part of this PR. I'm totally fine with the solution in https://github.com/Automattic/wp-calypso/commit/b7129a5603c62da5534497240679aaec332646a4 for now :)

Reply:
I did read it over a couple of times and my overarching priority is to not introduce any more complexity to the matrix we have going in `getTestAccountByFeature`, which is already quite complex from the looks of it.

To that end, since Jetpack tests are rather few and is not a priority for the Jetpack team - I think we can leave out the inclusion of `jetpack` feature type from the account matrix. 

I propose we continue witha simpler solution I have in https://github.com/Automattic/wp-calypso/commit/b7129a5603c62da5534497240679aaec332646a4 and make it abundantly clear that use of `TARGET_JETPACK` environment variable is an exception rather than the norm.

Reply:
Hrm, this one is a doozy... I do feel like the jetpack inclusion kind of muddies up the feature criteria to account table 😕 . It feels like we might be making that table way more complicated to handle something that has very narrow application.

I think I'm currently landing on...
1. We should still add `TARGET_JETPACK` to the `envVariables` to get all the good default behavior and typing.
2. We can avoid adding it to the feature tables for now. Instead, in the few scripts that need this, we can do something like...

```typescript
let accountName;

if ( envVariables.TARGET_JETPACK) {
    accountName = 'jetpackUser'
} else {
    const features = envToFeatureKey( envVariables );
    accountName = getTestAccountByFeature( features );
}
```

Reply:
> It feels like we might be making that table way more complicated to handle something that has very narrow application.
> I propose we continue witha simpler solution I have in https://github.com/Automattic/wp-calypso/commit/b7129a5603c62da5534497240679aaec332646a4 and make it abundantly clear that use of TARGET_JETPACK environment variable is an exception rather than the norm.

That makes a lot of sense! We can always revisit this later if we need more test site combos for JP. Thanks for sharing your thoughts!

Reply:
```
let accountName;

if ( envVariables.TARGET_JETPACK) {
    accountName = 'jetpackUser'
} else {
    const features = envToFeatureKey( envVariables );
    accountName = getTestAccountByFeature( features );
}
```

I will update the changes here to use this switching logic instead.